os/pm/pm_checkstate.c

```diff
@@ -100,51 +100,28 @@
 
 enum pm_state_e pm_checkstate(void)
 {
-	clock_t now;
 	irqstate_t flags;
 	int index;
-
-	/* Check for the end of the current time slice.  This must be performed
-	 * with interrupts disabled so that it does not conflict with the similar
-	 * logic in pm_activity().
-	 */
-
 	flags = enter_critical_section();
 	g_pmglobals.recommended = PM_STANDBY;
-
-	/* Check the elapsed time.  In periods of low activity, time slicing is
-	 * controlled by IDLE loop polling; in periods of higher activity, time
-	 * slicing is controlled by driver activity.  In either case, the duration
-	 * of the time slice is only approximate; during times of heavy activity,
-	 * time slices may be become longer and the activity level may be over-
-	 * estimated.
-	 */
-
-	now = clock_systimer();
-	if (now - g_pmglobals.stime >= TIME_SLICE_TICKS) {
-
-		/* Reset the time and recommended board to sleep.
-		 * This is an atomic operation because interrupts are still disabled.
-		 */
-
-		g_pmglobals.stime = now;
-		g_pmglobals.recommended = PM_SLEEP;
-
-	}
-
 	/* If there is power state lock for LCD and IDLE domain, recommended PM_NORMAL State */
 	if (g_pmglobals.suspend_count[PM_IDLE_DOMAIN] || g_pmglobals.suspend_count[PM_LCD_DOMAIN]) {
```

Comment:
It is require here also, because we are using auto-suspended timer, which will run at most one time extra before suspending itself. So if we don't keep this check here, timer will change LCD backlight irrespective of PM_LCD_DOMAIN is suspended or not